### PR TITLE
sdl12-compat: fix test on Linux

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

The versioned `libSDL` is `libSDL-1.2` on Linux.

### CODE REVIEW

**Comment by Bo98** [comment]
I'd also add something like this to the test:

```rb
(testpath/"test.c").write <<~EOS
  #include <SDL.h>

  int main() {
    SDL_Init(SDL_INIT_EVERYTHING);
    SDL_Quit();
    return 0;
  }
EOS
system ENV.cc, "test.c", "-I#{include}", "-L#{lib}", "-lSDL", "-o", "test"
system "./test"
```

Might need to swap `SDL_INIT_EVERYTHING` with `SDL_INIT_TIMER` or something if it doesn't like the video setup in the CI.

**Comment by carlocab** [comment]
Applied:
```diff
diff --git a/Formula/sdl12-compat.rb b/Formula/sdl12-compat.rb
index 94f8402a552..132ccb8fca1 100644
--- a/Formula/sdl12-compat.rb
+++ b/Formula/sdl12-compat.rb
@@ -35,5 +35,17 @@ class Sdl12Compat < Formula
     assert_predicate lib/shared_library(versioned_libsdl), :exist?
     assert_predicate lib/"libSDLmain.a", :exist?
     assert_equal version.to_s, shell_output("#{bin}/sdl-config --version").strip
+
+    (testpath/"test.c").write <<~EOS
+      #include <SDL/SDL.h>
+
+      int main() {
+        SDL_Init(SDL_INIT_EVERYTHING);
+        SDL_Quit();
+        return 0;
+      }
+    EOS
+    system ENV.cc, "test.c", "-I#{include}", "-L#{lib}", "-lSDL", "-o", "test"
+    system "./test"
   end
 end
```
but `brew test sdl12-compat` errors with:
```
❯ brew test --verbose sdl12-compat
==> Testing sdl12-compat
==> /usr/local/Cellar/sdl12-compat/1.2.52/bin/sdl-config --version
==> /usr/bin/clang test.c -I/usr/local/Cellar/sdl12-compat/1.2.52/include -L/usr/local/Cellar/sdl12-compat/1.2.52/lib -lSDL -o test
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
     (maybe you meant: _SDL_main)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: sdl12-compat: failed
```

**Comment by Bo98** [comment]
Ah yeah I think it needs `-lSDLmain` and stuff like that.

Try:

```rb
system ENV.cc, "test.c", "-o", "test", *Utils.safe_popen_read("sdl-config", "--cflags", "--libs").split
```

Might need to adjust the `#include` again as I think the `-I` flag from the config will include a trailing `/SDL`.

**Comment by carlocab** [comment]
ARM runners didn't like `SDL_INIT_EVERYTHING`, so trying `SDL_INIT_TIMER`.

**Comment by carlocab** [comment]
ARM fails with `Timeout::Error: execution expired`.

**Comment by Bo98** [comment]
Odd. Will take look in a bit.

Would be worthwhile testing if `SDL_INIT_EVERYTHING` works on Linux as that's where I expected it to fail.

**Comment by carlocab** [comment]
I think it works on Linux. Will push that change shortly.

**Comment by carlocab** [comment]
Yup, works on Linux.

**Comment by Bo98** [comment]
hmm

**Comment by Bo98** [comment]
Passing an RPATH to SDL2's lib directory for the test compile will probably work.

There's perhaps an argument we could do this by default: https://github.com/libsdl-org/sdl12-compat/blob/cc52691e4f32ad737f50a0a241e902a63fd7d15f/CMakeLists.txt#L252

but the `sdl` formula doesn't do that so maybe we don't need to. Not really looked at `sdl` dependents and if they work on arm64.

File: Formula/sdl12-compat.rb [hunk 1]
```diff
@@ -30,8 +30,22 @@ def install
...
+    (testpath/"test.c").write <<~EOS
+      #include <SDL.h>
+
+      int main() {
+        SDL_Init(SDL_INIT_EVERYTHING);
+        SDL_Quit();
+        return 0;
+      }
+    EOS
+    system ENV.cc, "test.c", "-o", "test", *Utils.safe_popen_read(bin/"sdl-config", "--cflags", "--libs").split
```
Comment (carlocab) [inline Formula/sdl12-compat.rb:48]:
```suggestion
    EOS
    flags = Utils.safe_popen_read(bin/"sdl-config", "--cflags", "--libs").split
    flags << "-Wl,-rpath,#{Formula["sdl2"].opt_lib}"
    system ENV.cc, "test.c", "-o", "test", *flags
```

**Comment by Bo98** [comment]
Seems like that broke the Linux test for some reason.

**Comment by carlocab** [comment]
Should be fixed now. Setting `SDL_RLD_FLAGS` might be good, but it looks like that'll have to be via an `inreplace`.

**Comment by Bo98** [comment]
Yeah, something to think about. I'm fine with this for now given it matches the current `sdl` behaviour.

**Review by Bo98 — APPROVED** [review]

**Comment by carlocab** [comment]
We might want to consider making this or `sdl` keg-only. Otherwise we may need to migrate everything that depends on `sdl` all at once because of the conflict.

**Comment by Bo98** [comment]
Yeah, I think there's no immediate rush to migrate. Might be worth letting a release or two go by first to fix any compat issues, since this is the first stable version release.

An option on the table could even be to outright remove `sdl` and point it to `sdl12-compat`.

**Comment by carlocab** [comment]
That makes sense. Would that works for things like `sdl_gfx`, `sdl_image`, etc?

Also, another option for the `RPATH` thing on ARM would be to just set `CMAKE_INSTALL_RPATH` to `Formula["sdl2"].opt_lib` during the build to avoid having to pass the right flags at all when linking.

**Comment by Bo98** [comment]
> Would that works for things like `sdl_gfx`, `sdl_image`, etc?

If this compat library works as intended: those would be able to depend on `sdl12-compat`.

> Also, another option for the `RPATH` thing on ARM would be to just set `CMAKE_INSTALL_RPATH` to `Formula["sdl2"].opt_lib` during the build to avoid having to pass the right flags at all when linking.

Ah yes, that might work.

**Comment by carlocab** [comment]
> > Would that works for things like `sdl_gfx`, `sdl_image`, etc?
> 
> 
> 
> If this compat library works as intended: those would be able to depend on `sdl12-compat`.

I was thinking more about the formulae that depend on `sdl_*`. I guess this means that while we can get rid of `sdl` itself, we'd still need to keep its derivatives around.

**Pull request merged**